### PR TITLE
extend Module to fix_param_names that saves gradient computation and update faster rcnn example

### PR DESCRIPTION
**Pull request by precedenceguo**
set_lr_mult and set_wd_mult are good choice for fine-tuning networks on other vision tasks. However gradients would be calculated even if weights were not to be updated. Adding `fix_param_names` to Module and DataParallelExecutorGroup will enable the binding process of executors to recognize trainable parameters.

Faster R-CNN demo is updated accordingly. More details will be finalized after the fix of cudnn auto tuning problem.

### CODE REVIEW

**Comment by piiswrong** [comment]
please fix lint and we can merge

**Comment by piiswrong** [comment]
Also the readme still says mxnet-detection fork. Do you still need it?

**Comment by precedenceguo** [comment]
No, there is no need for mxnet-detection fork anymore. I have an experiment ongoing to verify the training script going smooth. The example seems mixed but will be fixed to use module only as soon as the experiment ends.

**Comment by piiswrong** [comment]
Ok. Could you also update the readme?

**Comment by precedenceguo** [comment]
Experiment indicates comparable performance to py-faster-rcnn. README has been updated, removing mxnet-detection or old solver. Previous presented fast r-cnn training example has also adopted the new module api.

**Comment by piiswrong** [comment]
LGTM. @tqchen Could you merge this.
Let's post to reddit for some PR

**Comment by piiswrong** [comment]
could you update to master?

**Comment by precedenceguo** [comment]
Sure I just rebased to master.
